src/main/java/witchinggadgets/common/recipes/alchemic/WG_alchemic_clusters.java

```diff
@@ -2,265 +2,444 @@
+
+        @Optional.Method(modid = "gregtech_nh")
+        public Materials getGT5uMaterial() {
+            return Materials.getMaterialsMap().get(matName);
+        }
+
+        public ItemStack getPart(String prefix, int amount) {
+            return Utilities.getOredict(prefix + matName, amount);
+        }
+
+        @Optional.Method(modid = "gregtech_nh")
+        public ItemStack getPart(OrePrefixes prefix, int amount) {
+            return GTOreDictUnificator.get(prefix, getGT5uMaterial(), amount);
+        }
+
+        public String matName() {
+            return matName;
+        }
+
+        public MetaInfo meta() {
+            return meta;
+        }
+
+        public boolean ebf() {
+            return ebf;
+        }
+
+        public FluidStack liquid() {
+            return liquid;
+        }
+
+        public int vibrant() {
+            return vibrant;
+        }
+
+        public ClusterOverlay clusterOverlay() {
+            return clusterOverlay;
+        }
+    }
+
+    public static final Object2ObjectOpenHashMap<String, ClusterInfo> CLUSTER_INFO = new Object2ObjectOpenHashMap<>();
+    public static final Int2ObjectOpenHashMap<ClusterInfo> CLUSTER_INFO_BY_META = new Int2ObjectOpenHashMap<>();
+
     public static void registerClusters() {
+        if (Loader.isModLoaded("postea")) {
+            registerPosteaTransformer();
+        }
 
-        AspectList alchemyAspects;
+        if (WitchingGadgets.isGT5uLoaded) {
+            loadGT5uClusters();
+        } else {
+            loadMiscClusters();
+        }
 
-        if (Loader.isModLoaded("gregtech") && !Loader.isModLoaded("gregapi")) {
-            for (int iOre = 0; iOre < WGContent.GT_Cluster.length; iOre++) {
-                if (WGConfig.allowClusters) {
-                    ItemStack ingot = OreDictionary.getOres("ore" + WGContent.GT_Cluster[iOre]).get(0);
+        CLUSTER_INFO.forEach((name, clusterInfo) -> {
+            CLUSTER_INFO_BY_META.put(clusterInfo.meta.getMeta(), clusterInfo);
 
-                    if (ingot == null) {
-                        WitchingGadgets.logger.error(WGContent.GT_Cluster[iOre] + " == null! This should not happen!");
-                        continue;
-                    }
+            setupCluster(clusterInfo);
+        });
+    }
 
-                    try {
-                        alchemyAspects = ThaumcraftApi.objectTags
-                                .get(Arrays.asList(ingot.getItem(), ingot.getItemDamage())).add(Aspect.ORDER, 1);
-                    } catch (NullPointerException e) {
-                        WitchingGadgets.logger.error("Could not get the objectTags for" + WGContent.GT_Cluster[iOre]);
-                        alchemyAspects = new AspectList().add(Aspect.METAL, 2).add(Aspect.ORDER, 1)
-                                .add((Aspect) gregtech.api.enums.TCAspects.NEBRISUM.mAspect, 2);
-                    }
+    private static boolean clusterBlacklist(Materials material) {
```

Comment:
Doesn't this need a `@Optional.Method`?

```suggestion
    @Optional.Method(modid = "gregtech_nh")
    private static boolean clusterBlacklist(Materials material) {
```